### PR TITLE
Recommend non-zero values for footgun TPs

### LINKED ISSUE #1720: initial_max_stream_data_bidi_remote is a very capable foot gun
Draft 14 splits the definition of "initial_max_stream_data" into two parameters, "initial_max_stream_data_bidi_local(0)" and "initial_max_stream_data_bidi_remote(10)". In early interop tests, we observed an interesting failure to communicate between the picoquic client and the winquic server because of that. Winquic server did not provide a value for the initial_max_stream_data_bidi_remote parameter, which meant that for client initiated streams like stream 0 the initial value of was 0. As a result, picoquic did not send any data on stream zero, the stream zero was not open, winquic did not send back any update for the stream 0 MAX_STREAM_DATA value, the server did not receive any request from the client, and after a suitable period of silence the connection was closed.

I would suggest two fixes:

1) Explain that servers that expect to receive messages such as HTTP requests from clients SHOULD send the "initial_max_stream_data_bidi_remote(10)" parameter and set it to a reasonable value.

2) Explain that if a client wants to open a stream but initial_max_stream_data_bidi_remote value is 0, it SHOULD send an empty STREAM frame to open the stream -- and give the server a chance to send back MAX_STREAM_DATA update for that stream.

### PR DESCRIPTION
**Pull request by MikeBishop**
Fixes #1720.

### CODE REVIEW

File: draft-ietf-quic-http.md [hunk 1]
```diff
@@ -261,16 +261,23 @@ the stream management.
...
 responses.  A bidirectional stream ensures that the response can be readily
 correlated with the request. This means that the client's first request occurs
-on QUIC stream 0, with subsequent requests on stream 4, 8, and so on. HTTP/QUIC
-does not use server-initiated bidirectional streams. The use of unidirectional
-streams is discussed in {{unidirectional-streams}}.
+on QUIC stream 0, with subsequent requests on stream 4, 8, and so on. In order
+to permit these streams to open, an HTTP/QUIC client SHOULD send non-zero values
+for the QUIC transport parameters `initial_max_stream_data_bidi_local`. An
+HTTP/QUIC server SHOULD send non-zero values for the QUIC transport parameters
+`initial_max_stream_data_bidi_remote` and `initial_max_bidi_streams`.
```
Comment (kazuho) [inline draft-ietf-quic-http.md]: Would it make sense to recommend `initial_max_bidi_streams` to be "no smaller than 100", like what RFC 7540 states?

File: draft-ietf-quic-http.md [hunk 2]
```diff
@@ -261,16 +261,23 @@ the stream management.
...
 These streams carry frames related to the request/response (see {{frames}}).
 When a stream terminates cleanly, if the last frame on the stream was truncated,
 this MUST be treated as a connection error (see HTTP_MALFORMED_FRAME in
 {{http-error-codes}}).  Streams which terminate abruptly may be reset at any
 point in the frame.
 
+HTTP/QUIC does not use server-initiated bidirectional streams. The use of
+unidirectional streams is discussed in {{unidirectional-streams}}.  Both clients
+and servers SHOULD send a non-zero value for the QUIC transport parameter
+`initial_max_uni_streams`.
```
Comment (kazuho) [inline draft-ietf-quic-http.md]: Would it make sense to state recommend "at least three", considering the fact that there would be the controller stream and the QPACK encoder / decoder streams?

File: draft-ietf-quic-http.md (hunk 1 above)
Comment (mikkelfj) [reply draft-ietf-quic-http.md]: As I don't really follow HTTP/Q atm., I am not very confident in this - but if QPACK uses static tables only and you are on a constrained device, you would not want to track that much state.

Comment (kazuho) [reply draft-ietf-quic-http.md]: I do not oppose to your point.

What I am looking for is a general guidance that is similar to what we have in RFC 7540, quote: _”It is recommended that this value be no smaller than 100, so as to not unnecessarily limit parallelism.”_

**Pull request merged**